src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service;
+
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
```

Comment:
According to the encapsulation principle, consider making the 'operationStrategy' field private if it's not intended to be accessed directly from outside the class.